src/test/compile-fail/return-from-diverging.rs

```diff
@@ -8,9 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-// pretty-expanded FIXME #23616
+fn fail() -> ! {
+    return "wow"; //~ ERROR mismatched types
```

Comment:
I did delete some error messages in the process of implementing this RFC. There's no more "return in a function declared as diverging" error message for example.